lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/IMRUDriver.cs

```diff
@@ -98,181 +133,517 @@ internal sealed class IMRUDriver<TMapInput, TMapOutput, TResult, TPartitionType>
             [Parameter(typeof(MemoryPerMapper))] int memoryPerMapper,
             [Parameter(typeof(MemoryForUpdateTask))] int memoryForUpdateTask,
             [Parameter(typeof(AllowedFailedEvaluatorsFraction))] double failedEvaluatorsFraction,
+            [Parameter(typeof(MaxRetryNumberInRecovery))] int maxRetryNumberInRecovery,
             [Parameter(typeof(InvokeGC))] bool invokeGC,
-            IGroupCommDriver groupCommDriver)
+            IGroupCommDriver groupCommDriver,
+            INameServer nameServer)
         {
             _configurationManager = configurationManager;
-            _evaluatorRequestor = evaluatorRequestor;
             _groupCommDriver = groupCommDriver;
-            _coresPerMapper = coresPerMapper;
-            _coresForUpdateTask = coresForUpdateTask;
-            _memoryPerMapper = memoryPerMapper;
-            _memoryForUpdateTask = memoryForUpdateTask;
+            _nameServer = nameServer;
             _perMapperConfigs = perMapperConfigs;
             _totalMappers = dataSet.Count;
-
-            _allowedFailedEvaluators = (int)(failedEvaluatorsFraction * dataSet.Count);
             _invokeGC = invokeGC;
+            _maxRetryNumberForFaultTolerant = maxRetryNumberInRecovery != 0 ? maxRetryNumberInRecovery : DefaultMaxNumberOfRetryInRecovery;
```

Comment:
For better readability, please consider switching to the equality check in the ternary operator.